src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int START_VALUE = 0;
+    private static final int STEP = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyValue = calculateSupplyValue(readFromFile(fromFileName));
+        int buyValue = calculateBuyValue(readFromFile(fromFileName));
+        writeInfoToFile(supplyValue, buyValue, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+                new FileReader(new File(fromFileName)))) {
+            StringBuilder builder = new StringBuilder();
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                String[] listArray = line.split(",");
+                for (String list : listArray) {
+                    builder.append(list).append(",");
+                }
+                line = bufferedReader.readLine();
+            }
+            return builder.toString().split(",");
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file" + e);
+        }
+    }
 
+    private int calculateSupplyValue(String[] string) {
+        int supplyValue = START_VALUE;
+        for (int i = 0; i < string.length; i++) {
+            if (string[i].equals(SUPPLY)) {
+                supplyValue += Integer.parseInt(string[i + STEP]);
```

Comment:
Be careful with the 'STEP' constant. If the data format changes, relying on a fixed step to access the next value might lead to errors. It's safer to find the next integer after the 'supply' keyword programmatically.